ddl/ddl_api.go

```diff
@@ -2305,6 +2338,75 @@ func (d *ddl) AddColumn(ctx sessionctx.Context, ti ast.Ident, spec *ast.AlterTab
 	return errors.Trace(err)
 }
 
+// AddColumns will add multi new columns to the table.
+func (d *ddl) AddColumns(ctx sessionctx.Context, ti ast.Ident, specs []*ast.AlterTableSpec) error {
+	schema, t, err := d.getSchemaAndTableByIdent(ctx, ti)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	var columns []*table.Column
+	var positions []*ast.ColumnPosition
+	var offsets []int
+
+	// Check all the columns at once.
+	newColumnsCount := 0
+	addingColumnNames := make(map[string]bool)
+	for _, spec := range specs {
+		for _, specNewColumn := range spec.NewColumns {
+			newColumnsCount++
+			if !addingColumnNames[specNewColumn.Name.Name.O] {
+				addingColumnNames[specNewColumn.Name.Name.O] = true
+			} else {
+				return errors.Trace(infoschema.ErrColumnExists.GenWithStackByArgs(specNewColumn.Name.Name.O))
+			}
+		}
+	}
+	if err = checkAddColumnTooManyColumns(len(t.Cols()) + newColumnsCount); err != nil {
+		return errors.Trace(err)
+	}
+
+	// Check the columns one by one.
+	for _, spec := range specs {
+		for _, specNewColumn := range spec.NewColumns {
+			colName := specNewColumn.Name.Name.O
+			// Check whether added column has existed.
```

Comment:
Put line #2372 ~ 2381 into `checkAndCreateNewColumn ` function. and remove line #2306 ~ 2315.